pandas/core/dtypes/cast.py

```diff
@@ -1559,6 +1559,11 @@ def construct_1d_arraylike_from_scalar(
             dtype = np.dtype("object")
             if not isna(value):
                 value = ensure_str(value)
+        elif dtype.kind == "M" and is_valid_nat_for_dtype(value, dtype):
```

Comment:
should be ```dtype.kind in ['M', 'm']``` (do the tests fail w/o this?)

Reply:
no, they didn't before with "M" only (except for the ARM tests which I believe have an unrelated problem).

Trying with timedelta "m" now...